ccp.c

```diff
@@ -396,6 +414,7 @@ int datapath_program_install(struct DatapathProgram *datapath_programs, struct I
 
 }
 
+/* TODO unused? testing removal
```

Comment:
it used to be used on ccp connection close, but now datapath programs are "global". Datapath program list should maybe become some sort of LRU?

Reply:
hm yeah. or maybe at least "least recently installed", i.e. when it runs out of slots in the array it just loops back to the beginning.

in that case it probably makes more sense for the datapath to give the program an id rather than portus.. that way we don't need to loop through the entire program array each time to find the correct one, we can just jump directly to it

Reply:
the reason we didn't do this is it requires portus to get a response from libccp saying what the datapath program id is, so that change_program can work

this requires "connections" or at least "responses" that tie messages together, which is annoying to imlpement - currently the messages between portus and libccp are entirely standalone

anyway this is a separate concern, let's file an issue and come back to it?